remotes/docker/config/hosts.go

```diff
@@ -263,6 +263,23 @@ func hostDirectory(host string) string {
 	return host
 }
 
+// hostSuffix converts "docker.io" to ".docker.io" or "test.docker.io" to ".docker.io" or "localhost" to ".localhost"
+func hostSuffix(host string) string {
```

Comment:
oOo that's a good point, I forgot that .co.uk was a TLD

Reply:
How about including the suffix without the subdomain, so that this covers uri's with more than 3 segments. For example, "test.myregistry.co.uk" would match with ".myregistry.co.uk" and ".co.uk" folders?

Reply:
Ok better idea. According to IANA and Wikipedia, registrars only use the second level hierarchy for country codes. I could just create a set to check if the TLD is a country code, and then include that in the logic to pull the actual suffix out. That way I can preserve just the single match I'm looking for.

Reply:
Convenient list w/ a table of country codes that allow a second level to be registered. https://en.wikipedia.org/wiki/Country_code_top-level_domain

Reply:
Best to keep this logic as simple as possible, pulling in any knowledge related to tld complicates something that should be a simple match. It's not unreasonable to do a directory list here and perform pattern matching. `_` could easily fit in for a wildcard. Could also only support the wildcard at the beginning, such as `_.myregistry.io`. The logic is too restrictive if it assumes one possible wildcard match and starting a directory with `.` is not a good idea here for usability.